c7n/resources/ec2.py

```diff
@@ -898,6 +898,139 @@ def process_resource_set(self, client, resources):
             r[self.annotation] = info_map.get(r['InstanceId'], {})
 
 
+@filters.register('monitoring-status')
+class MonitoringStatus(ValueFilter):
+    """Filters EC2 Instances to get detailed monitoring status
+
+    Filtering relies on on valid values of monitoring statuses :
+    'disabled'|'disabling'|'enabled'|'pending'
+    (https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html#EC2.Client.describe_instances)
+
+    :Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: ec2-detailed-monitoring-activation
+            resource: ec2
+            filters:
+              - type: monitoring-status
+                key: State
+                value: disabled
+            actions:
+              - type: enable-monitoring
+    References
+
+     https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-cloudwatch-new.html
+    """
+    valid_values = ['disabled', 'disabling', 'enabled', 'pending']
+
+    schema = type_schema(
+        'monitoring-status',
+        rinherit=ValueFilter.schema,
+        value={'enum': valid_values},
+        required=('value',))
+    permissions = list('ec2:DescirbeInstances')
+
+    def process(self, resources, event=None):
+        client = utils.local_session(
+            self.manager.session_factory).client('ec2')
+        return [r for r in resources
+                if self.check_monitoring_status(client, r, self.data.get('value'))]
+
+    def check_monitoring_status(self, client, inst, value):
+        resp = client.describe_instances(
```

Comment:
why is this calling describe instances again? there really isn't any need for this filter afaics, the instance resource is the output of describe instances, and this filter is satisifed by the default value filter as an attribute check

for example

```
 - Monitoring.State: enabled
```